lib/rules/no-useless-backreference.js

```diff
@@ -72,11 +72,12 @@ module.exports = {
         schema: [],
 
         messages: {
-            nested: "Backreference '{{ bref }}' will be ignored. It references group '{{ group }}' from within that group.",
-            forward: "Backreference '{{ bref }}' will be ignored. It references group '{{ group }}' which appears later in the pattern.",
-            backward: "Backreference '{{ bref }}' will be ignored. It references group '{{ group }}' which appears before in the same lookbehind.",
-            disjunctive: "Backreference '{{ bref }}' will be ignored. It references group '{{ group }}' which is in another alternative.",
-            intoNegativeLookaround: "Backreference '{{ bref }}' will be ignored. It references group '{{ group }}' which is in a negative lookaround."
+            nested: "Backreference '{{ bref }}' will be ignored. It references group '{{ group }}'{{ otherGroups }} from within that group.",
+            forward: "Backreference '{{ bref }}' will be ignored. It references group '{{ group }}'{{ otherGroups }} which appears later in the pattern.",
+            backward: "Backreference '{{ bref }}' will be ignored. It references group '{{ group }}'{{ otherGroups }} which appears before in the same lookbehind.",
+            disjunctive: "Backreference '{{ bref }}' will be ignored. It references group '{{ group }}'{{ otherGroups }} which is in another alternative.",
+            intoNegativeLookaround: "Backreference '{{ bref }}' will be ignored. It references group '{{ group }}'{{ otherGroups }} which is in a negative lookaround.",
+            someReason: "Backreference '{{ bref }}' will be ignored. It references group '{{ group }}'{{ otherGroups }} which cannot be referenced for some reason."
```

Comment:
If there are multiple groups, with different reasons, the message doesn't specify any of the reasons?

Reply:
When there are multiple different reasons, all but one must be that the referenced groups are in other alternatives?

What do you think about the following:

1. If the backreference references only groups in other alternatives, then we report that as the problem.
2. Otherwise, we report the problem for the group that is in the same alternative. Maybe also include in the message that the backreference also references groups in other alternatives.

Reply:
Thank you for the review and comment. I like your idea :+1: I will change it to be like that.

Reply:
I have changed this rule, could you please check it again?